src/app/routes/onDemandRadio/getInitialData/index.js

```diff
@@ -62,26 +32,17 @@ const getEpisodeAvailableUntil = path([
   '0',
   'availableUntil',
 ]);
-const getReleaseDateTimeStamp = pathWithLogging(
-  ['metadata', 'releaseDateTimeStamp'],
-  { logLevel: LOG_LEVELS.WARN },
-);
-const getPromoBrandTitle = path(['promo', 'brand', 'title']);
-const getDurationISO8601 = pathWithLogging([
-  'promo',
-  'media',
-  'versions',
-  0,
-  'durationISO8601',
-]);
-const getThumbnailImageUrl = json =>
-  getPlaceholderImageUrlUtil(
-    pathWithLogging(['promo', 'media', 'imageUrl'])(json),
-  );
 
 export default async ({ path: pathname }) => {
   const onDemandRadioDataPath = overrideRendererOnTest(pathname);
   const { json, ...rest } = await fetchPageData(onDemandRadioDataPath);
+
+  // If no data was returned by fetchPageData, we can just skip all the data processing
```

Comment:
I understand why we should have a comment here because the error handling with the HOC is a little unintuitive. Should we then do this for all routes? If we do, should have a function that does called validateJson or something which contains this comment. If we do is it the best placed in all routes or in the fetchPageData function. I'm just thinking about because I have some unfinished work which kind of thinks about these things https://github.com/bbc/simorgh/pull/6852/files#diff-93bc8a6ae91fb1ee01233d35e3590c4d